lib/dal/src/func/authoring/save.rs

```diff
@@ -237,14 +236,7 @@ async fn update_attribute_associations(
     ctx: &DalContext,
     func: &Func,
     prototypes: Vec<AttributePrototypeBag>,
-    arguments: Vec<FuncArgumentBag>,
 ) -> FuncAuthoringResult<()> {
-    // First, modify the arguments because they dictate what we can do within the
-    // attribute subsystem.
-    save_attr_func_arguments(ctx, func, arguments).await?;
-
-    // Now that we know what func arguments exist and have been modified, we can work
-    // within the attribute subsystem.
     let backend_response_type =
         save_attr_func_prototypes(ctx, func, prototypes, true, None).await?;
```

Comment:
**This is the reason why this PR exist. The `arguments` and `prototypes` can get out of sync quickly. `FuncArguments` should be handled independently and then `prototypes` (which include `prototypeArguments`, what that are called in `app/web`) can use said `FuncArguments`.**

Reply:
So after picking up this PR - we needed to take this a step further as while we separated out the Func Arguments, we also needed to separate out how we handle prototype arguments. This was because when adding a new argument to  a shared attribute function, since we were still saving ALL prototype arguments for all schema variants, the saves would fail if you only configured the new prototype argument for ONE schema variant and not the others.  

I broke out CRUD routes for create/update/remove attribute prototypes so we can individually save each schema variants prototype without having to configure the prototypes for all schema variants at once. 

Note: the 'associations' for an attribute func are still returned when we get func details, I didn't break out separate list calls for expediency, but I expect this domain to evolve and didn't want to go to far in architecting to account for the current experience we provide.